content/docs/add-react-to-an-existing-app.md

```diff
@@ -6,91 +6,92 @@ prev: add-react-to-a-new-app.html
+```js
+// Render in several places
 
-The smallest React example looks like this:
+ReactDOM.render(
+  <Button label="Sign Out" />,
+  document.querySelector('.sign-out')
+);
 
-```js
-import React from 'react';
-import ReactDOM from 'react-dom';
+ReactDOM.render(
+  <Button label="Sign In" />,
+  document.querySelector('.sign-in')
+);
 
 ReactDOM.render(
-  <h1>Hello, world!</h1>,
-  document.getElementById('root')
+  <Button label="Edit" />,
+  document.querySelector('.edit')
 );
 ```
 
-This code renders into a DOM element with the id of `root`, so you need `<div id="root"></div>` somewhere in your HTML file.
+### 3. Compile with Babel
 
-Similarly, you can render a React component inside a DOM element somewhere inside your existing app written with any other JavaScript UI library.
+[Babel](https://babeljs.io) is a tool that converts [JSX](/docs/introducing-jsx.html) and future JavaScript syntax into ES5 JavaScript compatible with older browsers.
 
-[Learn more about integrating React with existing code.](/docs/integrating-with-other-libraries.html#integrating-with-other-view-libraries)
+_Install `babel` with [`yarn`](https://yarnpkg.com) or [`npm`](https://npmjs.com)_
 
-### A Complete Example
+```shell
+npm init # if you don't already have a package.json
+npm install --global babel-cli
+npm install babel-preset-react-app
+```
 
-You can find step-by-step instructions detailing a basic implementation from scratch, including Babel and Webpack setup [here](https://medium.com/@JedaiSaboteur/creating-a-react-app-from-scratch-f3c693b84658).
+_Example: compile files in `src` folder to `build`_
 
-### Development and Production Versions
+```shell
+NODE_ENV=development
+babel --presets=react-app src -d build
+```
 
-By default, React includes many helpful warnings. These warnings are very useful in development.
+### 4. Add components to page
 
-**However, they make the development version of React larger and slower so you should use the production version when you deploy the app.**
+```html
+<script src="build/button.js"></script>
+```
 
-Learn [how to tell if your website is serving the right version of React](/docs/optimizing-performance.html#use-the-production-build), and how to configure the production build process most efficiently:
+### 5. Production
 
-* [Creating a Production Build with Create React App](/docs/optimizing-performance.html#create-react-app)
-* [Creating a Production Build with Single-File Builds](/docs/optimizing-performance.html#single-file-builds)
-* [Creating a Production Build with Brunch](/docs/optimizing-performance.html#brunch)
-* [Creating a Production Build with Browserify](/docs/optimizing-performance.html#browserify)
-* [Creating a Production Build with Rollup](/docs/optimizing-performance.html#rollup)
-* [Creating a Production Build with webpack](/docs/optimizing-performance.html#webpack)
+You probably already have a minification step for your JavaScript - don't forget to minify your component files too!
```

Comment:
don't need this, if you are using the prod version of react ( I think)

Reply:
The  component scripts won't minify themselves 😄 
The production CDN links are minified copies of React & ReactDOM though